File: ce/customer-engagement/on-premises/snippets/visualbasic/CRMV8/relationshipsvb/vb/workwithrelationships1.vb
' <snippetworkwithrelationships1>



 Dim eligibleCreateOneToManyRelationship As Boolean =
  Me.EligibleCreateOneToManyRelationship("account", "campaign")

 If eligibleCreateOneToManyRelationship Then
  Dim createOneToManyRelationshipRequest As CreateOneToManyRequest =
   New CreateOneToManyRequest With {
    .OneToManyRelationship = New OneToManyRelationshipMetadata With {
     .ReferencedEntity = "account",
     .ReferencingEntity = "campaign",
     .SchemaName = "new_account_campaign",
     .AssociatedMenuConfiguration =
     New AssociatedMenuConfiguration With {
      .Behavior = AssociatedMenuBehavior.UseLabel,
      .Group = AssociatedMenuGroup.Details,
      .Label = New Label("Account", 1033),
      .Order = 10000},
     .CascadeConfiguration =
     New CascadeConfiguration With {
      .Assign = CascadeType.Cascade,
      .Delete = CascadeType.Cascade,
      .Merge = CascadeType.Cascade,
      .Reparent = CascadeType.Cascade,
      .Share = CascadeType.Cascade,
      .Unshare = CascadeType.Cascade
     }
    },
    .Lookup = New LookupAttributeMetadata With {
     .SchemaName = "new_parent_accountid",
     .DisplayName = New Label("Account Lookup", 1033),
     .RequiredLevel = New AttributeRequiredLevelManagedProperty(AttributeRequiredLevel.None),
     .Description = New Label("Sample Lookup", 1033)
    }
   }


  Dim createOneToManyRelationshipResponse As CreateOneToManyResponse =
   CType(_serviceProxy.Execute(createOneToManyRelationshipRequest), CreateOneToManyResponse)

  _oneToManyRelationshipId = createOneToManyRelationshipResponse.RelationshipId
  _oneToManyRelationshipName = createOneToManyRelationshipRequest.OneToManyRelationship.SchemaName

  Console.WriteLine("The one-to-many relationship has been created between {0} and {1}.", "account", "campaign")
 End If

' </snippetworkwithrelationships1>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/relationships/cs/workwithrelationships1.cs
// <snippetworkwithrelationships1>



 bool eligibleCreateOneToManyRelationship =
     EligibleCreateOneToManyRelationship("account", "campaign");

 if (eligibleCreateOneToManyRelationship)
 {
     CreateOneToManyRequest createOneToManyRelationshipRequest =
         new CreateOneToManyRequest
     {
         OneToManyRelationship =
         new OneToManyRelationshipMetadata
         {
             ReferencedEntity = "account",
             ReferencingEntity = "campaign",
             SchemaName = "new_account_campaign",
             AssociatedMenuConfiguration = new AssociatedMenuConfiguration
             {
                 Behavior = AssociatedMenuBehavior.UseLabel,
                 Group = AssociatedMenuGroup.Details,
                 Label = new Label("Account", 1033),
                 Order = 10000
             },
             CascadeConfiguration = new CascadeConfiguration
             {
                 Assign = CascadeType.NoCascade,
                 Delete = CascadeType.RemoveLink,
                 Merge = CascadeType.NoCascade,
                 Reparent = CascadeType.NoCascade,
                 Share = CascadeType.NoCascade,
                 Unshare = CascadeType.NoCascade
             }
         },
         Lookup = new LookupAttributeMetadata
         {
             SchemaName = "new_parent_accountid",
             DisplayName = new Label("Account Lookup", 1033),
             RequiredLevel = new AttributeRequiredLevelManagedProperty(AttributeRequiredLevel.None),
             Description = new Label("Sample Lookup", 1033)
         }
     };


     CreateOneToManyResponse createOneToManyRelationshipResponse =
         (CreateOneToManyResponse)_serviceProxy.Execute(
         createOneToManyRelationshipRequest);

     _oneToManyRelationshipId =
         createOneToManyRelationshipResponse.RelationshipId;
     _oneToManyRelationshipName = 
         createOneToManyRelationshipRequest.OneToManyRelationship.SchemaName;

     Console.WriteLine(
         "The one-to-many relationship has been created between {0} and {1}.",
         "account", "campaign");
 }

// </snippetworkwithrelationships1>